### PR TITLE
Reorganized client library section.

### PR DESCRIPTION
**Pull request by ralph-lange**

### CODE REVIEW

File: _docs/concepts/client_library/diagnostics/index.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+title: Diagnostics
+permalink: /docs/concepts/client_library/diagnostics/
+---
+
+Although diagnostics is not part of the ROS 2 Client Library packages (i.e., rclcpp, rclpy), it definitely can be counted to the extended client library as it provides very generic and application-independent functionalities.
+
+That's why the micro-ROS Client Library also comes with basic diagnostics functionalities. These are fully compatible with ROS 2 diagnostics, but comprise three features only:
```
Comment (norro) [inline _docs/concepts/client_library/diagnostics/index.md:8]: Wrt. _"fully compatible"_:
I obviously had to introduce new message types without dynamics arrays that require separate aggregation/treatment, e.g., in the uros agent (as you indicated in the diagnostic_architecture illustration). Is that what you are referring to? 
As the diagnostic aggregator does not natively work with our uros diagnostic msgs but requires this level of indirection, I would go for a weaker phrase, e.g., _compatible_ or _integrated with_.

File: _docs/concepts/client_library/features/index.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+title: Feature List
+permalink: /docs/concepts/client_library/features/
+---
+
+The micro-ROS Client Library, formed by standard [ROS 2 Client Support Library (rcl)](https://github.com/ros2/rcl/) and the new [ROS 2 Client Library package (rclc)](https://github.com/ros2/rclc/), is going feature all major ROS concepts such as
```
Comment (norro) [inline _docs/concepts/client_library/features/index.md:6]: ...is going ***to*** feature...

File: _docs/concepts/client_library/introduction/index.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+title: Introduction to Client Library
+permalink: /docs/concepts/client_library/introduction/
+redirect_from:
+  - /docs/concepts/
+  - /docs/concepts/client_library/
+---
+
+The Client Library provides the micro-ROS API for the user code, i.e., for application-level micro-ROS nodes. The goal is to provide all relevant, major ROS 2 concepts in an optimized implementation for microcontrollers. At the same time, we strive to make the API as compatible as possible to standard ROS 2, to facilitate porting of code.
+
+To minimize the long-term maintenance cost, we use existing data structures and algorithms from the ROS 2 stack and to bring necessary changes in the mainline stack as far as possible. That's why the micro-ROS client library is built up from standard [ROS 2 Client Support Library (rcl)](https://github.com/ros2/rcl/) and a new [ROS 2 Client Library package (rclc)](https://github.com/ros2/rclc/). Together, as depicted below, rcl + rclc form a feature-complete client library in the C programming language.
```
Comment (norro) [inline _docs/concepts/client_library/introduction/index.md:11]: "and to bring necessary"-> "and bring necessary"

**Review by norro — DISMISSED** [review]
Nur kleine Anmerkungen

File: _docs/concepts/client_library/diagnostics/index.md (hunk 1 above)
Comment (ralph-lange) [inline _docs/concepts/client_library/diagnostics/index.md:8]:
```suggestion
That's why the micro-ROS Client Library also comes with basic diagnostics functionalities. These are compatible with ROS 2 diagnostics, but comprise three features only:
```

File: _docs/concepts/client_library/introduction/index.md (hunk 3 above)
Comment (ralph-lange) [inline _docs/concepts/client_library/introduction/index.md:11]:
```suggestion
To minimize the long-term maintenance cost, we use existing data structures and algorithms from the ROS 2 stack and bring necessary changes in the mainline stack as far as possible. That's why the micro-ROS client library is built up from standard [ROS 2 Client Support Library (rcl)](https://github.com/ros2/rcl/) and a new [ROS 2 Client Library package (rclc)](https://github.com/ros2/rclc/). Together, as depicted below, rcl + rclc form a feature-complete client library in the C programming language.
```

File: _docs/concepts/client_library/features/index.md (hunk 2 above)
Comment (ralph-lange) [inline _docs/concepts/client_library/features/index.md:6]:
```suggestion
The micro-ROS Client Library, formed by standard [ROS 2 Client Support Library (rcl)](https://github.com/ros2/rcl/) and the new [ROS 2 Client Library package (rclc)](https://github.com/ros2/rclc/), is going to feature all major ROS concepts such as
```

**Review by norro — APPROVED** [review]

**Review by norro — APPROVED** [review]
LGTM

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**